test/unit/mixins/evented.test.js

```diff
@@ -0,0 +1,455 @@
+
+  assert.throws(
+    function() {
+      target.on();
+    },
+    new Error('invalid event type; must be a non-empty string or array')
+  );
+
+  assert.throws(
+    function() {
+      target.on('   ');
+    },
+    new Error('invalid event type; must be a non-empty string or array')
+  );
+
+  assert.throws(
+    function() {
+      target.on([]);
+    },
+    new Error('invalid event type; must be a non-empty string or array'),
+    ''
+  );
+
+  assert.throws(
+    function() {
+      target.on('x');
+    },
+    new Error('invalid listener; must be a function')
+  );
+
+  assert.throws(
+    function() {
+      target.on({}, 'x', () => {});
+    },
+    new Error('invalid target; must be a DOM node or evented object')
+  );
+
+  assert.throws(
+    function() {
+      target.on(evented({}), 'x', null);
+    },
+    new Error('invalid listener; must be a function')
+  );
+});
+
+QUnit.test('one() errors', function(assert) {
+  const target = this.targets.a = evented({});
+
+  assert.throws(
+    function() {
+      target.one();
+    },
+    new Error('invalid event type; must be a non-empty string or array')
+  );
+
+  assert.throws(
+    function() {
+      target.one('   ');
+    },
+    new Error('invalid event type; must be a non-empty string or array')
+  );
+
+  assert.throws(
+    function() {
+      target.one([]);
+    },
+    new Error('invalid event type; must be a non-empty string or array'),
+    ''
+  );
+
+  assert.throws(
+    function() {
+      target.one('x');
+    },
+    new Error('invalid listener; must be a function')
+  );
+
+  assert.throws(
+    function() {
+      target.one({}, 'x', () => {});
+    },
+    new Error('invalid target; must be a DOM node or evented object')
+  );
+
+  assert.throws(
+    function() {
+      target.one(evented({}), 'x', null);
+    },
+    new Error('invalid listener; must be a function')
+  );
+});
+
+QUnit.test('off() errors', function(assert) {
+  const target = this.targets.a = evented({});
+
+  // An invalid event actually causes an invalid target error because it
+  // gets passed into code that assumes the first argument is the target.
+  assert.throws(
+    function() {
+      target.off([]);
+    },
+    new Error('invalid target; must be a DOM node or evented object')
+  );
+
+  assert.throws(
+    function() {
+      target.off({}, 'x', () => {});
+    },
+    new Error('invalid target; must be a DOM node or evented object')
+  );
+
+  assert.throws(
+    function() {
+      target.off(evented({}), '', () => {});
+    },
+    new Error('invalid event type; must be a non-empty string or array')
+  );
+
+  assert.throws(
+    function() {
+      target.off(evented({}), [], () => {});
+    },
+    new Error('invalid event type; must be a non-empty string or array')
+  );
+
+  assert.throws(
+    function() {
+      target.off(evented({}), 'x', null);
+    },
+    new Error('invalid listener; must be a function')
+  );
+});
+
+QUnit.test('a.on("x", fn)', function(assert) {
```

Comment:
I suppose it's the style used in most places. I've still never really drank the BDD Kool-aid entirely. 😄